src/compose/images.ts

```diff
@@ -52,584 +47,599 @@ type NormalisedDockerImage = Docker.ImageInfo & {
 	NormalisedRepoTags: string[];
 };
 
-export class Images extends (EventEmitter as new () => ImageEventEmitter) {
-	public appUpdatePollInterval: number;
-
-	private imageFetchFailures: Dictionary<number> = {};
-	private imageFetchLastFailureTime: Dictionary<
-		ReturnType<typeof process.hrtime>
-	> = {};
-	private imageCleanupFailures: Dictionary<number> = {};
-	// A store of volatile state for images (e.g. download progress), indexed by imageId
-	private volatileState: { [imageId: number]: Image } = {};
-
-	public constructor() {
-		super();
+// Setup an event emitter
+interface ImageEvents {
+	change: void;
+}
+class ImageEventEmitter extends (EventEmitter as new () => StrictEventEmitter<
+	EventEmitter,
+	ImageEvents
+>) {}
+const events = new ImageEventEmitter();
+
+export const on: typeof events['on'] = events.on.bind(events);
+export const once: typeof events['once'] = events.once.bind(events);
+export const removeListener: typeof events['removeListener'] = events.removeListener.bind(
+	events,
+);
+export const removeAllListeners: typeof events['removeAllListeners'] = events.removeAllListeners.bind(
+	events,
+);
+
+const imageFetchFailures: Dictionary<number> = {};
+const imageFetchLastFailureTime: Dictionary<ReturnType<
+	typeof process.hrtime
+>> = {};
+const imageCleanupFailures: Dictionary<number> = {};
+// A store of volatile state for images (e.g. download progress), indexed by imageId
+const volatileState: { [imageId: number]: Image } = {};
+
+let appUpdatePollInterval: number;
+
+export const initialized = (async () => {
+	await config.initialized;
+	appUpdatePollInterval = await config.get('appUpdatePollInterval');
+	config.on('change', (vals) => {
+		if (vals.appUpdatePollInterval != null) {
+			appUpdatePollInterval = vals.appUpdatePollInterval;
+		}
+	});
+})();
+
+export async function triggerFetch(
+	image: Image,
+	opts: FetchOptions,
+	onFinish = _.noop,
+	serviceName: string,
+): Promise<null> {
```

Comment:
Consider returning `Promise<void>` because triggerFetch is a command rather then query and the return type is the give away for the difference. 

```suggestion
): Promise<void> {
```